core/services/user_service/controllers/auth_controller.py

```diff
@@ -42,3 +42,33 @@ def login(email: str, password: str) -> Tuple[str, str]:
     })
 
     return access_token, refresh_token
+
+
+def refresh_access_token(refresh_token: str):
+    """
+    The refresh function takes a refresh token
+    and generates a new access_token.
+    Returns a tuple of access_token and refresh_token
+    """
+    payload = verify_token(refresh_token)
+
+    if not payload:
+        raise HTTPException(status_code=401, detail="Invalid refresh token")
+
+    email: str = payload.get("email")
+    
+    db: Session = next(get_database())
+    user: User = db.query(User).filter_by(email=email).first()
+
+    if not user:
+        raise HTTPException(404, detail="User not found")
```

Comment:
This should be 403 instead of 404. Username enumeration possible

Reply:
Is "Username enumeration possible" a proposed detail?